File: chapter 2/learning_service/src/add_client.cpp
#include<ros/ros.h>
#include<learning_service/Add.h>
#include<cstdlib>

int main(int argc, char **argv)
{
    ros::init(argc ,argv, "add_client");

    if (argc != 3)
    {
        ROS_ERROR("usage: add two number X Y");
        return 1;
    }
    ros::NodeHandle nh;

    ros::ServiceClient add_client = nh.serviceClient<learning_service::Add>("add_service");

    learning_service::Add srv;
    srv.request.a = atof(argv[1]);
    srv.request.b = atof(argv[2]);

    if(add_client.call(srv))
    {
        ROS_INFO("sum is: %0.2f", srv.response.sum);
    }
    else
    {
        ROS_ERROR("Failed to call service add_service");
        return 1;
    }

    return 0;
}

File: chapter 2/learning_service/scripts/add_client.py
#!/usr/bin/env python
# coding:utf-8

import sys
import rospy
from learning_service.srv import Add

def add_client(x, y):
    rospy.init_node('add_client')
    rospy.wait_for_service('add_service')
    try:
        add_two_ints = rospy.ServiceProxy('add_service', Add)
        response = add_two_ints(x, y)
        print("sum is: %0.2f" %response.sum)
    except rospy.ServiceException as e:
        print("Failed to call service add_service: %s"%e)

if __name__ == "__main__":
    if len(sys.argv) == 3:
        x = float(sys.argv[1])
        y = float(sys.argv[2])
        add_client(x, y)
    else:
        print("usage: add two number X Y")
        sys.exit(1)